gateway/controller.go

```diff
@@ -178,8 +178,16 @@ func (c *ControllerHTTPClient) SetNamespace(namespace string) {
 }
 
 // FetchAllSchemas fetches all schemas
-func (c *ControllerHTTPClient) FetchAllSchemas() (tables []common.Table, err error) {
-	return c.GetAllSchema(c.namespace)
+func (c *ControllerHTTPClient) FetchAllSchemas() (tables []*common.Table, err error) {
+	schemas, err := c.GetAllSchema(c.namespace)
```

Comment:
here since you are using :=, the err will shadow the err in the return arguments, so anyway you will return a nil err.
I recommend you either using non-named return arguments or pre declare schemas

Reply:
BTW, I think you need to write a test case to catch above non-nil error even it fails

Reply:
good catch